File: src/controllers/auth.controller.ts
import { Response, NextFunction } from "express";
import { AuthenticatedRequest } from "../types";
import { AuthService } from "../services/auth.service";

export class AuthController {
  static async createSuperAdmin(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const result = await AuthService.createSuperAdmin(req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(400).json(result);
    } catch (error) {
      next(error);
    }
  }

  static async register(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const result = await AuthService.registerSchool(req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(400).json(result);
    } catch (error) {
      next(error);
    }
  }

  static async login(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const result = await AuthService.login(req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(401).json(result);
    } catch (error) {
      next(error);
    }
  }

  static async refreshToken(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const result = await AuthService.refreshToken(req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(401).json(result);
    } catch (error) {
      next(error);
    }
  }

  static async changePassword(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const userId = req.user?.id;
      if (!userId) {
        return res.status(401).json({ success: false, message: "Unauthorized context" });
      }
      const result = await AuthService.changePassword(userId, req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(400).json(result);
    } catch (error) {
      next(error);
    }
  }
}

File: _old_csharp_backup/LeonEdBackend/Controllers/AuthController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using LeonEdBackend.DTOs.AuthDtos;
using LeonEdBackend.DTOs.Common;
using LeonEdBackend.Services.Interfaces;

namespace LeonEdBackend.Controllers
{
    [Route("api/[controller]")]
    public class AuthController : BaseController
    {
        private readonly IAuthService _authService;
        public AuthController(IAuthService authService) { _authService = authService; }

        /// <summary>Create the platform Super Admin (one-time setup, protected by secret key)</summary>
        [HttpPost("create-super-admin")]
        [AllowAnonymous]
        public async Task<IActionResult> CreateSuperAdmin([FromBody] CreateSuperAdminRequest request)
        {
            var result = await _authService.CreateSuperAdmin(request);
            return result.Success ? Ok(result) : BadRequest(result);
        }

        /// <summary>Register a new school with an admin account</summary>
        [HttpPost("register")]
        [AllowAnonymous]
        public async Task<IActionResult> Register([FromBody] RegisterSchoolRequest request)
        {
            var result = await _authService.RegisterSchool(request);
            return result.Success ? Ok(result) : BadRequest(result);
        }

        /// <summary>Login with email and password</summary>
        [HttpPost("login")]
        [AllowAnonymous]
        public async Task<IActionResult> Login([FromBody] LoginRequest request)
        {
            var result = await _authService.Login(request);
            return result.Success ? Ok(result) : Unauthorized(result);
        }

        /// <summary>Refresh an expired JWT token</summary>
        [HttpPost("refresh-token")]
        [AllowAnonymous]
        public async Task<IActionResult> RefreshToken([FromBody] RefreshTokenRequest request)
        {
            var result = await _authService.RefreshToken(request);
            return result.Success ? Ok(result) : Unauthorized(result);
        }

        /// <summary>Change the current user's password</summary>
        [HttpPost("change-password")]
        [Authorize]
        public async Task<IActionResult> ChangePassword([FromBody] ChangePasswordRequest request)
        {
            var result = await _authService.ChangePassword(GetUserId(), request);
            return result.Success ? Ok(result) : BadRequest(result);
        }
    }
}
